pangeo_forge_recipes/storage.py

```diff
@@ -171,19 +171,26 @@ def _full_path(self, path: str) -> str:
         return os.path.join(self.root_path, new_path)
 
 
+@dataclass
 class CacheFSSpecTarget(FlatFSSpecTarget):
     """Alias for FlatFSSpecTarget"""
 
+    verify_existing: bool = True
+
     def cache_file(self, fname: str, secrets: Optional[dict], **open_kwargs) -> None:
         # check and see if the file already exists in the cache
         logger.info(f"Caching file '{fname}'")
-        if self.exists(fname):
+        exists = self.exists(fname)
+        if exists and self.verify_existing:
             cached_size = self.size(fname)
             remote_size = _get_url_size(fname, secrets, **open_kwargs)
             if cached_size == remote_size:
                 # TODO: add checksumming here
```

Comment:
For future reference, do we have a way of doing this without downloading the file and computing the checksum? S3 provides that information but arbitrary HTTP endpoints can't be relied upon to provide the SHA in a predictable way

Reply:
OK, so digging into fsspec source a bit, it appears that fsspec's `info` method ought to be used here. It will, by default, avoid opening the file and instead attempt to get both size and checksum from a `head` request. It looks to me as though there's no benefit to caching with the current method if `verify_existing` is `True`, as it will get the size of an opened file rather than using a small `HEAD` request

Would you like me to amend this PR with some logic to lean on those methods? Probably good to provide a warning in case it is absolutely necessary to fully load the file up (as happens here) for size/checksum validation